File: backend/src/schemas/schedule.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime
from src.schemas.user import UserPublic
from src.schemas.transaction import TransactionRead


class ScheduleCreate(BaseModel):
    group_id: int
    user_id: int
    title: str
    date: datetime
    location: Optional[str] = None
    description: Optional[str] = None


class ScheduleCalendarRead(BaseModel):
    id: int
    title: str
    date: datetime
    is_done: bool
    location: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class AllScheduleCalendarRead(BaseModel):
    id: int
    title: str
    date: datetime
    is_done: bool
    group_id: int
    group_name: str
    location: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ScheduleCommentRead(BaseModel):
    id: int
    user: UserPublic
    content: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class ScheduleImageRead(BaseModel):
    id: int
    image_url: str
    uploaded_at: datetime

    model_config = ConfigDict(from_attributes=True)

class ScheduleRead(BaseModel):
    id: int
    group_id: int
    diary_id: Optional[int] = None
    user: UserPublic
    title: str
    images: Optional[list[ScheduleImageRead]] = None
    date: datetime
    location: Optional[str]
    is_done: bool
    description: Optional[str]
    created_at: datetime
    comments: list[ScheduleCommentRead] = []
    transactions: list[TransactionRead] = []

    model_config = ConfigDict(from_attributes=True)


class ScheduleUpdate(BaseModel):
    title: Optional[str] = None
    date: Optional[datetime] = None
    location: Optional[str] = None
    description: Optional[str] = None
    is_done: Optional[bool] = None


class ScheduleCommentCreate(BaseModel):
    user_id: int
    content: str

class ScheduleImageCreate(BaseModel):
    schedule_id: int
    image_url: str

class PendingScheduleRead(BaseModel):
    id: int
    group_id: int
    title: str
    date: datetime
    is_done: bool
    location: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/schedule.ts
// src/types/schedule.ts

import { TransactionRead } from "./transaction";

export interface ScheduleItem {
  id: number;
  title: string;
  date: string; // ISO 문자열, ex) "2025-05-03T11:50:49.169000"
  is_done: boolean;

  location?: string; // 추가
  description?: string; // 추가
}

export interface ScheduleImage {
  id: number;
  image_url: string;
  uploaded_at: string;
}

export interface ScheduleComment {
  id: number;
  user: {
    id: number;
    name: string;
    profile_image_url?: string | null;
  };
  content: string;
  created_at: string;
}

export interface ScheduleDetail extends ScheduleItem {
  group_id: number;
  user: {
    id: number;
    name: string;
    profile_image_url?: string | null; // ✅ 추가
  };
  location?: string;
  description?: string;
  created_at: string;
  comments: ScheduleComment[];
  transactions: TransactionRead[]; // ✅ 이 줄 추가!
  diary_id?: number | null; // ✅ 이 줄 추가
  images?: ScheduleImage[]; // ✅ 이 줄 추가
}

export interface ScheduleCreateInput {
  user_id: number;
  title: string;
  date: string; // "YYYY-MM-DD"
  time: string; // "HH:mm"
  location?: string;
  description?: string;
}

export interface ScheduleCardItem {
  id: number;
  title: string;
  date: string;
  is_done: boolean;
  location?: string;
}

export interface AllScheduleCardItem extends ScheduleCardItem {
  group_id: number;
  group_name: string;
}

export interface PendingSchedule {
  id: number;
  group_id: number;
  title: string;
  date: string;
  is_done: boolean;
  location?: string;
}
